ffcx/codegeneration/ufc.h

```diff
@@ -120,40 +104,43 @@ extern "C"
     /// Return the family of the finite element function space
     const char* family;
 
-    int (*evaluate_reference_basis)(double* restrict reference_values,
-                                    int num_points, const double* restrict X);
-
-    int (*evaluate_reference_basis_derivatives)(
-        double* restrict reference_values, int order, int num_points,
-        const double* restrict X);
-
     /// Map the reference values on to the cell
     /// @param[in] cell_permutations An integer that says how each entity of the
     ///         cell of dimension < tdim has been permuted relative to a
     ///         low-to-high ordering of the cell.
-
     int (*transform_reference_basis_derivatives)(
         double* restrict values, int order, int num_points,
         const double* restrict reference_values, const double* restrict X,
         const double* restrict J, const double* restrict detJ,
-        const double* restrict K, const uint32_t cell_permutation);
-
-    /// Map values of field from physical to reference space which has
-    /// been evaluated at points given by
-    /// tabulate_reference_dof_coordinates.
-    int (*transform_values)(ufc_scalar_t* restrict reference_values,
-                            const ufc_scalar_t* restrict physical_values,
-                            const double* restrict coordinate_dofs,
-                            const ufc_coordinate_mapping* restrict cm);
-
-    // FIXME: change to 'const double* reference_dof_coordinates()'
-    /// Tabulate the coordinates of all dofs on a reference cell
-    int (*tabulate_reference_dof_coordinates)(
-        double* restrict reference_dof_coordinates);
+        const double* restrict K);
+
+    /// Apply dof tranformations to some data
+    /// @param[in] data The data to be transformed
+    /// @param[in] cell_permutation An integer encoding the orientation of the
+    /// cell's entities
+    /// @param[in] dim The number of data items for each DOD
+    int (*apply_dof_transformation)(double* data, uint32_t cell_permutation,
```

Comment:
For more complicated situations such as face tangents, the DOF transformation is not a permutation, as it combines multiple DOFs to get the new DOF. It's probably more sensible though to call the functions permutations everywhere though to avoid confusion, so I'll rename stuff

Reply:
Actually, just noticed that there's `apply_dof_transformation` and `get_dof_permutation` and I guess it's useful to have different names to distinguish these